File: LengthOfLongestPalindrome.java
/**
 * Approach : Maintain a DataStructer to put a character's count or itself.
 *            if it occurs twice increase the count for palindrome by 2
 *            at the end check if there are any elements remaining or not, if not empty return count increased by 1 for middle element 
 * 
 * TC : O(n) SC : O(1) because the set of characters will be limited 
 */
class Solution {
    public int longestPalindrome(String s) {
        if(s.length() == 0) return 0;
        HashSet<Character> set = new HashSet<>();
        int count = 0;
        for(int i = 0; i < s.length(); i++){
            char c = s.charAt(i);
            if(set.contains(c)){
                set.remove(c);
                count+=2;
            }else {
                set.add(c);
            } 
            
        }
        if(!set.isEmpty()) count++;
        return count;
    }
}

File: lengthOfLongestPalindrome.py
# TC: O(n) looping through each character
# SC: O(1) in worst case size will be 26
# LC: Yes
class Solution:
    def longestPalindrome(self, s: str) -> int:
        hashmap = dict()
        for c in s:
            if c not in hashmap:
                hashmap[c] = 1
            else:
                del hashmap[c]
        if len(hashmap) > 1:
            return len(s) - len(hashmap) +1
        return len(s)